system/metrics/testing/metrictest.cpp

```diff
@@ -64,13 +64,26 @@ const char *testConfigYml = R"!!(component:
 )!!";
 
 
+const char *testLogSinkConfigYml = R"!!(component:
+  metrics:
+    name: config_name
+    prefix: component_prefix.
+    sinks:
+        sink:
+          - type: log
+            name: default
+            settings:
+              period: 5
+)!!";
+
+
 int main(int argc, char *argv[])
 {
     InitModuleObjects();
 
     //
     // Simulate retrieving the component and global config
-    Owned<IPropertyTree> pSettings = createPTreeFromYAMLString(testConfigYml, ipt_none, ptr_ignoreWhiteSpace, nullptr);
+    Owned<IPropertyTree> pSettings = createPTreeFromYAMLString(testFileSinkConfigYml, ipt_none, ptr_ignoreWhiteSpace, nullptr);
```

Comment:
couldn't tell, but is this only built/packaged in DEBUG mode?

Reply:
Conditionally exclude the entire testing subproject based on "${CMAKE_BUILD_TYPE}"

Reply:
I don't think it is a problem for testing code to be built in release mode - it is often useful.

Reply:
(provided it does not significantly increase the size of the deployed system.)  This is separate so it wouldn't.

Reply:
@ghalliday how do you envision this being used in "release"? I can see perhaps as part of smoke test or regression tests, but I still contend this and code like it shouldn't be packaged in our official release akin to Java junit tests which are built with the project, executed if desired, but not part of the target package.

Reply:
Well it should be using the cppunit framework that is used by all the other unit tests.  (Which is deployed on a separate option.)  The unit tests are often run in release as well as debug mode.  I agree it shouldn't be deployed as part of our standard release.